### PR TITLE
Leosight RPC request - fix

### PR DESCRIPTION
**Pull request by Dzardys**

### CODE REVIEW

**Review by Bas950 — CHANGES_REQUESTED** [review]
upload proof
dont upload js file

**Comment by Dzardys** [comment]
Better?

File: websites/L/Leosight/presence.ts [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+  }
+}
+
+//neco//
+if (presenceData.details === null) {
+  presence.setTrayTitle("Provádí nespecifikovanou činnost.");
+  presence.setActivity();
+} else 
+  presence.setActivity(presenceData);
+
```
Comment (EncryptedDev) [inline websites/L/Leosight/presence.ts:455]:
```suggestion
  presence.setActivity(presenceData);
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Hello Dzardys, 

- Fix CodeFactor issues;
- Instead of defining `presenceData.startTimestamp = time` over and over, define it once instead

File: websites/L/Leosight/presence.ts (hunk 1 above)
Comment (Dzardys) [reply websites/L/Leosight/presence.ts]: Well every time I use beautifier, it comes with more curly problems

Comment (EncryptedDev) [reply websites/L/Leosight/presence.ts]: No, that's not removing the curlies, it is just refactored code

Comment (Dzardys) [reply websites/L/Leosight/presence.ts]: Well, idk what to do, I have formatted the text for X times and there's still the problem (Closing curly brace does not appear on the same line as the subsequent block. (brace-style)). I'll better close the request, I'm getting bullied by curly braces.

File: websites/L/Leosight/presence.ts [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "lslogo",
+      smallImageKey: "lsminilogo",
+      smallImageText: "leosight.cz"
+    },
+    path = document.location.pathname;
+  presenceData.startTimestamp = time;
+
+  //filtrace fóra//
```
Comment (EncryptedDev) [inline websites/L/Leosight/presence.ts:15]: What does this mean? Is it neccessary?

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
- Fix CodeFactor issues
- Remove useless comments 
- If you set `presenceData.startTimestamp` to nothing, it will do nothing...

File: websites/L/Leosight/presence.ts (hunk 2 above)
Comment (Dzardys) [reply websites/L/Leosight/presence.ts:15]: It defines presenceData.startTimestamp to value time, which is defined in the 2nd line of code, i think. But it isn't necessary to have variable time, I can define it into presenceData.StartTimestamp.

Comment (EncryptedDev) [reply websites/L/Leosight/presence.ts:15]: Obviously I know what that does, but I was talking about L15

File: websites/L/Leosight/presence.ts [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+          else */
+        presenceData.state = "Vyhledává nemovitost";
+      } else if (path.includes("firma")) {
+        /* if(path.includes("/firma/")){
+            item = document.querySelector("div.container.shadow.bg-light.my-5.px-4.py-3 h1") as HTMLElement
+            presenceData.state = "Firma " + item.innerText
+          }
+          else */
+        presenceData.state = "Vyhledává firmu";
+      } else if (path.includes("smazane"))
```
Comment (Slowlife01) [inline websites/L/Leosight/presence.ts:305]: Remove these comments

File: websites/L/Leosight/presence.ts [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+        presenceData.state = "Sazebník trestů";
+      else if (path.includes("stop"))
+        presenceData.state = "Zastavovací techniky";
+      else if (path.includes("law")) presenceData.state = "Zákon o policii";
+      else if (path.includes("directive")) presenceData.state = "Směrnice";
+      else if (path.includes("teams")) presenceData.state = "Hierarchie PSA";
+    }
+    ////
+    //stoongle//
+    else if (path.includes("stoongle")) {
```
Comment (Slowlife01) [inline websites/L/Leosight/presence.ts:321]:
```suggestion
    } else if (path.includes("stoongle")) {
```

File: websites/L/Leosight/presence.ts [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+      presenceData.details = "Stoongluje";
+      presenceData.startTimestamp;
+      if (path.includes("katalog")) {
+        presenceData.state = "Katalog webů";
+        presenceData.startTimestamp;
+      }
+    }
+    ////
+    else if (path.includes("burza.ic")) {
```
Comment (Slowlife01) [inline websites/L/Leosight/presence.ts:330]:
```suggestion
    } else if (path.includes("burza.ic")) {
```

File: websites/L/Leosight/presence.ts [hunk 6]
```diff
@@ -0,0 +1,459 @@
...
+      presenceData.startTimestamp;
+    } else if (path.includes("krypta.ic")) {
+      presenceData.state = "Krypta - online kasíno";
+      presenceData.startTimestamp;
+    } else if (path.includes("katastr.ic")) {
+      presenceData.state = "Katastr nemovitostí";
+      presenceData.startTimestamp;
+    }
+    //lemon//
+    else if (path.includes("lemongate.ic")) {
```
Comment (Slowlife01) [inline websites/L/Leosight/presence.ts:350]:
```suggestion
    } else if (path.includes("lemongate.ic")) {
```

File: websites/L/Leosight/presence.ts [hunk 7]
```diff
@@ -0,0 +1,459 @@
...
+      presenceData.startTimestamp;
+    } else if (path.includes("irc4na.lemon")) {
+      presenceData.state = "LemonGate - Ircana";
+      presenceData.startTimestamp;
+    } else if (path.includes("1trade.lemon")) {
+      presenceData.state = "LemonGate - AceTrade";
+      presenceData.startTimestamp;
+    }
+    ////
+    else if (path.includes("blockchain.ic")) {
```
Comment (Slowlife01) [inline websites/L/Leosight/presence.ts:370]:
```suggestion
    } else if (path.includes("blockchain.ic")) {
```

File: websites/L/Leosight/presence.ts [hunk 8]
```diff
@@ -0,0 +1,459 @@
...
+      presenceData.startTimestamp;
+    } else if (path.includes("spay.ic")) {
+      presenceData.state = "SPay";
+      presenceData.startTimestamp;
+    } else if (path.includes("sherwood.ic")) {
+      presenceData.state = "Sherwood Corporation";
+      presenceData.startTimestamp;
+    }
+    //smail//
+    else if (path.includes("smail.ic")) {
```
Comment (Slowlife01) [inline websites/L/Leosight/presence.ts:390]:
```suggestion
    } else if (path.includes("smail.ic")) {
```

File: websites/L/Leosight/presence.ts [hunk 9]
```diff
@@ -0,0 +1,459 @@
...
+    } else if (path.includes("vaos.ic")) {
+      presenceData.state = "Veterinární a odchytová služba";
+      presenceData.startTimestamp;
+    } else if (path.includes("writer.ic")) {
+      presenceData.state = "Writer";
+      presenceData.startTimestamp;
+    }
+  }
+  //ctf//
+  else if (window.location.href.includes("ctf")) {
```
Comment (Slowlife01) [inline websites/L/Leosight/presence.ts:415]:
```suggestion
  } else if (window.location.href.includes("ctf")) {
```

File: websites/L/Leosight/presence.ts [hunk 10]
```diff
@@ -0,0 +1,459 @@
...
+    presenceData.smallImageKey = "ctf";
+    presenceData.smallImageText = "ctf.leosight.cz";
+    presenceData.details = "Leosight CTF";
+    presenceData.startTimestamp;
+    if (path.includes("login.php")) presenceData.state = "Přihlášení";
+    else if (path.includes("index.php"))
+      presenceData.state = "Odvařuje si mozek";
+  }
+  //eco//
+  else if (window.location.href.includes("eco")) {
```
Comment (Slowlife01) [inline websites/L/Leosight/presence.ts:425]:
```suggestion
  } else if (window.location.href.includes("eco")) {
```

File: websites/L/Leosight/presence.ts [hunk 11]
```diff
@@ -0,0 +1,459 @@
...
+    presenceData.smallImageKey = "eco";
+    presenceData.smallImageText = "eco.leosight.cz";
+    presenceData.details = "Hraje Leosight ECO";
+    presenceData.startTimestamp;
+  }
+  //guard//
+  else if (window.location.href.includes("guard")) {
```
Comment (Slowlife01) [inline websites/L/Leosight/presence.ts:432]:
```suggestion
  } else if (window.location.href.includes("guard")) {
```

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

File: websites/L/Leosight/presence.ts (hunk 11 above)
Comment (Dzardys) [reply websites/L/Leosight/presence.ts:432]: Thanks

File: websites/L/Leosight/presence.ts (hunk 2 above)
Comment (Dzardys) [reply websites/L/Leosight/presence.ts:15]: Ah, it was just for me to know, it isn't neccessary.

**Comment by Slowlife01** [comment]
Remove `presenceData.startTimestamp;`

File: websites/L/Leosight/presence.ts [hunk 12]
```diff
@@ -0,0 +1,413 @@
...
+  time = Math.floor(Date.now() / 1000);
+let item, item2;
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "lslogo",
+      smallImageKey: "lsminilogo",
+      smallImageText: "leosight.cz"
+    },
+    path = document.location.pathname;
+  presenceData.startTimestamp = time;
```
Comment (Slowlife01) [inline websites/L/Leosight/presence.ts:13]:
```suggestion
      smallImageText: "leosight.cz",
      startTimestamp: time
    },
    path = document.location.pathname;
```

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

File: websites/L/Leosight/presence.ts (hunk 12 above)
Comment (Dzardys) [reply websites/L/Leosight/presence.ts:13]: then I'll be able to remove it from everywhere without changing the function?

File: websites/L/Leosight/presence.ts [hunk 13]
```diff
@@ -0,0 +1,413 @@
...
+
+  if (
+    path === "/" &&
+    !window.location.href.includes("artic") &&
+    !window.location.href.includes("ctf") &&
+    !window.location.href.includes("eco") &&
+    !window.location.href.includes("guard")
+  ) {
+    presenceData.details = "Prohlíží si hlavní stránku...";
+    presenceData.startTimestamp;
```
Comment (Bas950) [inline websites/L/Leosight/presence.ts:25]: what are all these empty startTimestamps?

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/L/Leosight/presence.ts (hunk 12 above)
Comment (Slowlife01) [reply websites/L/Leosight/presence.ts:13]: yes?

Comment (Dzardys) [reply websites/L/Leosight/presence.ts:13]: K, thanx

**Comment by Dzardys** [comment]
Well, I hope it was the last change. It's my first time using TS and GitHub, my apologies for spamming commits and questions.

**Review by Slowlife01 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**